protocol/lavaprotocol/finalization_consensus.go

```diff
@@ -188,62 +217,51 @@ func (fc *FinalizationConsensus) NewEpoch(epoch uint64) {
 			utils.LavaFormatDebug("finalization information epoch changed", utils.Attribute{Key: "specId", Value: fc.specId}, utils.Attribute{Key: "epoch", Value: epoch})
 		}
 		// means it's time to refresh the epoch
-		fc.prevEpochProviderHashesConsensus = fc.currentProviderHashesConsensus
-		fc.currentProviderHashesConsensus = []ProviderHashesConsensus{}
+		fc.prevEpochBlockToHashesToAgreeingProviders = fc.currentBlockToHashesToAgreeingProviders
```

Comment:
better to work with pointers. more performant than creating a shallow copy every time.

Reply:
Not relevant anymore